x-pack/plugins/enterprise_search/public/applications/workplace_search/components/shared/source_row/source_row.tsx

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import classNames from 'classnames';
+import _kebabCase from 'lodash/kebabCase';
+import { Link } from 'react-router-dom';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSwitch,
+  EuiSwitchEvent,
+  EuiTableRow,
+  EuiTableRowCell,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+
+import { SOURCE_STATUSES as statuses } from '../../../constants';
+import { IContentSourceDetails } from '../../../types';
+import { ADD_SOURCE_PATH, SOURCE_DETAILS_PATH, getContentSourcePath } from '../../../routes';
+
+import { SourceIcon } from '../source_icon';
+
+const CREDENTIALS_INVALID_ERROR_REASON = 1;
+
+export interface ISourceRow {
+  showDetails?: boolean;
+  isOrganization?: boolean;
+  onSearchableToggle?(sourceId: string, isSearchable: boolean): void;
+}
+
+interface ISourceRowProps extends ISourceRow {
+  source: IContentSourceDetails;
+}
+
+export const SourceRow: React.FC<ISourceRowProps> = ({
+  source: {
+    id,
+    serviceType,
+    searchable,
+    supportedByLicense,
+    status,
+    statusMessage,
+    name,
+    documentCount,
+    isFederatedSource,
+    errorReason,
+    allowsReauth,
+  },
+  onSearchableToggle,
+  isOrganization,
+  showDetails,
+}) => {
+  const isIndexing = status === statuses.INDEXING;
+  const hasError = status === statuses.ERROR || status === statuses.DISCONNECTED;
+  const showFix =
+    isOrganization && hasError && allowsReauth && errorReason === CREDENTIALS_INVALID_ERROR_REASON;
+
+  const rowClass = classNames(
+    'source-row',
+    { 'content-section--disabled': !searchable },
+    { 'source-row source-row--error': hasError }
+  );
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
Why do we need to disable linter here?

Reply:
If not this error occurs:
![image](https://user-images.githubusercontent.com/1869731/91996828-0c930a00-ecff-11ea-85c0-5c35d821979f.png)

Reply:
Makes sense 👍